source/developer/dev-setup-centos-7.rst

```diff
@@ -0,0 +1,98 @@
+.. _dev-setup-centos-7:
+
+Installing Developer Components on CentOS 7
+===========================================
+
+Set up your development environment for building, running, and testing Mattermost.
+
+1. If you are developing with the Docker container, install and configure Docker CE:
+
+  a. Follow the Docker installation instructions at https://docs.docker.com/engine/installation/linux/docker-ce/centos/.
+
+  b. Edit your ``/etc/hosts`` file to include the following line:
+
+    ``127.0.0.1     dockerhost``
+
+  c. Add your username to the *docker* group so that you can run Docker without using sudo. Replace *{username}* with your username.
+
+    ``sudo gpasswd -a {username} docker``
+
+  d. Restart the Docker daemon.
+
+    ``sudo service docker restart``
+
+  e. Change your current group ID to the *docker* group.
+
+    ``newgrp docker``
+
+2. Install the development tools package and libpng12 required by pngquant(a library used by Mattermost).
+
+    a. ``sudo yum group install "Development Tools"``
+
+    b. ``sudo yum install -y libpng12``
+
+3. Download and install Go 1.8 for Linux:
+
+    a. Download the Go binary.
+
+       ``wget https://storage.googleapis.com/golang/go1.8.linux-amd64.tar.gz``
+
+    b. Install the Go binary.
+
+       ``sudo tar -C /usr/local -xzf go1.8.linux-amd64.tar.gz``
+
+4. Set up your Go workspace:
+
+  a. ``mkdir ~/go``
+
+  b. Add the following lines to your ``~/.bashrc`` file:
+
+    .. code-block:: bash
+
+      export GOPATH=$HOME/go
+      export PATH=$PATH:$GOPATH/bin
+      export PATH=$PATH:/usr/local/go/bin
+
+  c. Reload your bash configuration.
+
+      source ~/.bashrc
+
+  d. Since you cannot run ``ulimit -n 8096`` with a regular user, we have to edit the file ``sudo nano /etc/security/limits.conf`` to include the following 2 lines at the end of it:
```

Comment:
I don't know if that happens on Ubuntu, I can try later and let u know, but on CentOS 7 you cannot set that feature as a user on the default installation without doing it as a root, or change the settings like that. And the ulimit AFAIK is important to mattermost even though I am not sure it is needed for a dev setup unless they are also using it for load tests among other things that might trigger the nofile limit.